src/components/Heading/Heading.tsx

```diff
@@ -172,20 +181,60 @@ export const Heading = chakra(
             }
           : { fontSize: `mobile.heading.${finalRoot}${sizeIndex}` }
         : undefined;
-      return (
-        <ChakraHeading
-          as={asHeading}
-          className={className}
-          id={id}
-          ref={ref}
-          sx={{
-            ...styles,
-            ...responsiveStyles,
-          }}
-          {...rest}
-        >
-          {content}
-        </ChakraHeading>
+      const overlineSize = !isNaN(sizeIndex)
+        ? sizeIndex <= 2
+          ? "overline1"
+          : "overline2"
+        : undefined;
+      const subtitleSize = !isNaN(sizeIndex)
+        ? sizeIndex <= 2
+          ? "subtitle1"
+          : "subtitle2"
+        : undefined;
+      const wrapperStyles = styles.headingWrapper;
+      const headingStyles =
+        overline || subtitle
+          ? {
+              ...styles,
+              ...responsiveStyles,
+            }
+          : {
+              ...styles,
+              ...responsiveStyles,
+              ...wrapperStyles,
+            };
+      const finalContent = (
+        <>
+          {overline && (
+            <Text mb="xxs" size={overlineSize}>
+              {overline}
+            </Text>
+          )}
+          <ChakraHeading
+            as={asHeading}
+            className={className}
+            id={id}
+            ref={ref}
+            sx={{
+              ...headingStyles,
+            }}
+            {...rest}
+          >
+            {content}
+          </ChakraHeading>
+          {subtitle && (
+            <Text mt="xs" noSpace size={subtitleSize}>
+              {subtitle}
+            </Text>
+          )}
+        </>
+      );
+      return overline || subtitle ? (
+        <Box as="hgroup" sx={{ ...wrapperStyles }} {...rest}>
```

Comment:
Per @EdwinGuzman:
This would be good to discuss in the doc's "Accessibility" section, i.e. why or how hgroup helps in the case of a heading with either overline or subtitle or both.